From 72e56d63958f323476b16f69ccd38ed09a8fa3e7 Mon Sep 17 00:00:00 2001
From: David Seifert <soap@gentoo.org>
Date: Tue, 1 May 2018 14:21:02 +0200
Subject: [PATCH 14/16] Allow `required : false` for OpenMP dependency

* Currently `required : true` is implicitly assumed, making
  optional use of OpenMP not possible.
---
 mesonbuild/dependencies/misc.py | 8 +++++++-
 1 file changed, 7 insertions(+), 1 deletion(-)

diff --git a/mesonbuild/dependencies/misc.py b/mesonbuild/dependencies/misc.py
index d4525b11..cf84123d 100644
--- a/mesonbuild/dependencies/misc.py
+++ b/mesonbuild/dependencies/misc.py
@@ -253,7 +253,13 @@ class OpenMPDependency(ExternalDependency):
         language = kwargs.get('language')
         super().__init__('openmp', environment, language, kwargs)
         self.is_found = False
-        openmp_date = self.compiler.get_define('_OPENMP', '', self.env, [], [self])
+        try:
+            openmp_date = self.compiler.get_define('_OPENMP', '', self.env, [], [self])
+        except mesonlib.EnvironmentException as e:
+            mlog.debug('OpenMP support not available in the compiler')
+            mlog.debug(e)
+            openmp_date = False
+
         if openmp_date:
             self.version = self.VERSIONS[openmp_date]
             if self.compiler.has_header('omp.h', '', self.env, dependencies=[self]):
-- 
2.17.0

